zebra-script/src/lib.rs

```diff
@@ -74,8 +82,13 @@ pub struct CachedFfiTransaction {
 }
 
 impl CachedFfiTransaction {
-    /// Construct a `PrecomputedTransaction` from a `Transaction`.
-    pub fn new(transaction: Arc<Transaction>) -> Self {
+    /// Construct a `PrecomputedTransaction` from a `Transaction` and the outputs
+    /// from previous transactions that match each input in the transaction
+    /// being verified.
+    pub fn new(
+        transaction: Arc<Transaction>,
+        all_previous_outputs: Vec<transparent::Output>,
+    ) -> Self {
         let tx_to = transaction
```

Comment:
Can we rename this buffer to `tx_to_serialized`, to make it clear that it isn't a shared reference to any `self.transaction` data?